.gitlab-ci.yml

```diff
@@ -121,28 +122,52 @@ build-dependencies:
   variables:
     IMAGE_TAG: "datafed/dependencies"
     GIT_STRATEGY: clone
+    DATAFED_HARBOR_IMAGE_TAG: "$CI_COMMIT_SHA"
   tags:
     - docker
   script:
     - BRANCH_LOWER=$(echo "$CI_COMMIT_REF_NAME" | tr '[:upper:]' '[:lower:]')
     - docker system prune -f
     - docker build -f docker/Dockerfile.dependencies -t "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}:latest" .
+    - docker tag "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}:latest" "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}:$CI_COMMIT_SHA"
     - docker login "${REGISTRY}" -u "${HARBOR_USER}" -p "${HARBOR_DATAFED_GITLAB_CI_REGISTRY}"
-    - docker push "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}:latest"
+    - export DATAFED_HARBOR_REPOSITORY="${IMAGE_TAG}-${BRANCH_LOWER}"
+    - export DATAFED_HARBOR_USERNAME="${HARBOR_USER}"
+    - export DATAFED_HARBOR_PASSWORD="${HARBOR_DATAFED_GITLAB_CI_REGISTRY}"
+    - |
+      while [ "$(./scripts/ci_harbor_artifact_count.sh)" == "0" ]; do
+        echo "Artifact missing from harbor..."
+        docker push "${REGISTRY}/${DATAFED_HARBOR_REPOSITORY}:latest"
+        docker push "${REGISTRY}/${DATAFED_HARBOR_REPOSITORY}:$CI_COMMIT_SHA"
+        sleep 5  # Optional: Add a sleep to avoid busy waiting
```

Comment:
Lots of duplicate code, can we follow the idea [here](https://github.com/ORNL/DataFed/pull/966#discussion_r1679638826) with a template and [extends](https://docs.gitlab.com/ee/ci/yaml/#extends)?

Reply:
Yes, that was addressed in #966